driver-core/src/main/com/mongodb/internal/ClientSideOperationTimeout.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.internal;
+
+import com.mongodb.lang.Nullable;
+
+import static com.mongodb.assertions.Assertions.assertNotNull;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+/**
+ * Client Side Operation Timeout.
+ *
+ * <p>Includes support for the deprecated {@code maxTimeMS}, {@code maxAwaitTimeMS} and {@code maxCommitTimeMS} operation configurations</p>
+ */
+public class ClientSideOperationTimeout {
+
+    private final Long timeoutMS;
+
+    // Deprecated operation based operation timeouts
+    private final long maxTimeMS;
+    private final long maxAwaitTimeMS;
```

Comment:
`maxAwaitTimeMS` isn't actually deprecated - this will be fixed as part of J4058